src/nix/key.rs

```diff
@@ -100,9 +102,21 @@ impl Key {
                 Ok(Box::new(Cursor::new(content)))
             }
             KeySource::Command(command) => {
-                let pathname = &command[0];
+                let mut pathname = command[0].clone();
                 let argv = &command[1..];
 
+                // if the executable is a derivation, realize it and replace it with its output
```

Comment:
Good catch, but we probably should perform the build earlier in the process so the deployment doesn't stall at "Uploading keys" for a long while. We should also rely on the Hive's builder as well, which may not be local in the future.